### PR TITLE
Try to add GraalPython to CI

### PR DESCRIPTION
**Pull request by da-woods**
I have a list of tests that fail locally (although it may not be complete).
Enough seems to work that I think it's worth testing properly.

Only issue may be that it's pretty slow to run locally and uses a decent amount of CPU and memory.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -59,6 +59,12 @@ jobs:
 
           # Ubuntu sub-jobs:
           # ================
+          # graalpy is really slow, so put it first...
+          - os: ubuntu-20.04
+            python-version: pypy-3.10
```
Comment (da-woods) [inline .github/workflows/ci.yml:64]:
```suggestion
```

**Comment by da-woods** [comment]
It's on the absolute edge of being able to complete in the time available... I could increase the timeout, but 1 hr 20 is already quite long.

I did ask about performance at https://github.com/oracle/graalpython/issues/411 (since their claim is that their JIT should be better than CPython, but the biggest slowdown is running Cython itself)

**Comment by da-woods** [comment]
Just going to rerun this to see how much difference the `operator.getattrs` change makes...

**Comment by da-woods** [comment]
This is as working as I think it's going to be.

I'm a bit on the fence about whether this is worth the very long run time.

**Comment by da-woods** [comment]
There might be some benefit to running with the option `--vm.XX:ActiveProcessorCount=1` - on my laptop it restricts it to 2 (!?) cores and makes it slightly slower. But it'd let multiprocessing do its job more efficiently.

Even better is `--experimental-options --engine.Compilation=false` which turns off the JIT compiler completely. That restricts it to 1 core, and actually reduces the wall-clock time.

I'd need to work out if there's a good way of applying that reliably everywhere though.

**Comment by da-woods** [comment]
I think there's a few tests that are occasionally flaky, but we can deal with them if they cause regular problems.

Disabling the JIT (and re-enabling full parallelism) gets it down to ~40 mins, which I think is OK. So I'll merge this.

**Comment by scoder** [comment]
>Disabling the JIT (and re-enabling full parallelism) gets it down to ~40 mins, 

We already knew from PyPy that Visitor.py is not JIT friendly. Disabling it seems reasonable. Also, you probably already disabled the Cython compilation, right?

**Comment by da-woods** [comment]
> Also, you probably already disabled the Cython compilation, right?

Yes - no Cython compilation.

The other obvious potential time saving I can see is that it spends 5 minutes compiling Numpy at the start, which could probably be eliminated with caching. But I think I'll treat it as good enough for now.

**Pull request merged**